### PR TITLE
Get users books list + get only free books for non-admin user

### PR DESCRIPTION
**Pull request by Julia-Protasevich**

### CODE REVIEW

File: controllers/books.controller.js [hunk 1]
```diff
@@ -30,23 +45,34 @@ export const BooksController = {
...
 		try {
+			const userId = req.body.userId; //should I get it from session l8r?
+			let updateBookOptions;
+			if(userId) {					//if userID was provided, we book the book for this user
+				updateBookOptions = {
+					isTakenByUser: userId
+				};
+			} else {						//else it means that user wants to release the book, 
+				updateBookOptions = {		//so we remove the corresponding field value
+					$unset: {isTakenByUser: ""}
```
Comment (MihailMartinovich) [inline controllers/books.controller.js:56]: use single quotes for JS

File: controllers/books.controller.js [hunk 2]
```diff
@@ -30,23 +45,34 @@ export const BooksController = {
 
 	async updateBook(req, res, next) {
 		try {
+			const userId = req.body.userId; //should I get it from session l8r?
+			let updateBookOptions;
+			if(userId) {					//if userID was provided, we book the book for this user
```
Comment (MihailMartinovich) [inline controllers/books.controller.js:50]: you miss empty line between logically different sections of code.

File: controllers/books.controller.js [hunk 3]
```diff
@@ -30,23 +45,34 @@ export const BooksController = {
...
-				}
-			});
+			await Book.findByIdAndUpdate(
+				req.params.id, 
+				updateBookOptions, 
+				{
+					new: true,
+					returnNewDocument: true,
+				},
+				function (err, result) {
```
Comment (MihailMartinovich) [inline controllers/books.controller.js:67]: why are you using callback instead of async/await statement?

**Review by MihailMartinovich — CHANGES_REQUESTED** [review]

File: controllers/books.controller.js (hunk 1 above)
Comment (Julia-Protasevich) [reply controllers/books.controller.js]: Replaced with single quotes

File: controllers/books.controller.js (hunk 2 above)
Comment (Julia-Protasevich) [reply controllers/books.controller.js:51]: empty line added

File: controllers/books.controller.js (hunk 3 above)
Comment (Julia-Protasevich) [reply controllers/books.controller.js]: No more callback

**Pull request merged**